lib/cookie-support.ts

```diff
@@ -54,6 +61,13 @@ export class CookieInterceptor {
     return config;
   }
 
+  /**
+   * This is called by Axios when a 2xx response has been received.
+   * We'll invoke the configured cookie jar's setCookie() method to handle
+   * the "set-cookie" header.
+   * @param response the Axios response object
+   * @returns the response object
+   */
   public async responseInterceptor(response: AxiosResponse) {
     logger.debug('CookieInterceptor: intercepting response.');
     if (response && response.headers) {
```

Comment:
if the typing is correct, then this check is not needed, since `response` is guaranteed to exist and so is `response.headers`.

Reply:
We still have many users that use this package with vanilla JS. I get that the Axios API probably guarantees this but it still may be prudent to check and prevent unnecessary reference exceptions. What do you think?

Reply:
Just my 2cents: the response interceptor is used in the scope of `axios.interceptors.response` and the typing here is:

```ts
response: AxiosInterceptorManager<AxiosResponse>;
```
So axios guarantees that the interceptor is called with an `AxiosResponse`, the interceptor would not be called by a user directly, only via axios.

In such cases I rely on the contract provided by the library. To me it's one of the main advantages of TS to have these guarantees at compile time, so we do not need to check at runtime.

Reply:
That is fair - I'll make that change